wip-tmpolaczyk-unhandled-intercept-v2.md

```diff
@@ -0,0 +1,84 @@
+
+## Motivation and rationale
+
+During the implementation of [WIP-0017][WIP-0017], we have noticed that renaming a `RadError` may invalidate some tally transactions, creating a hard fork. The reason for that is the `UnhandledIntercept` error which has one extra argument, the "message", which is implemented in Rust as:
+
+```rust
+format!("inner: {:?}", inner)
+```
+
+In other words, it creates a string that contains the debug representation of the inner error. The debug representation is created by the Rust compiler, and it may change between Rust versions. And the inner error is supposed to be an implementation detail of witnet-rust, so it doesn't make much sense for it to be part of consensus-critial code. This gives us a good reason to remove this message.
+
+
+## Specification
+
+Starting from and including the activation epoch:
+
+**(1)**: All the data requests that result in an `UnhandledIntercept` RADON error (error code 0xFF) must serialize this error with no extra arguments.
+
+**(2)**: Any tally that contains an `UnhandledIntercept` RADON error (error code 0xFF) with arguments will be considered invalid.
+
+
+## Backwards compatibility
+
+- Implementer: a client that implements or adopts the protocol improvements proposed in this document.
+- Non-implementer: a client that fails to or refuses to implement the protocol improvements proposed in this document.
+
+
+#### Consensus cheat sheet
+
+Upon entry into force of the proposed improvements:
+
+- Blocks and transactions that were considered valid by former versions of the protocol MUST continue to be considered valid.
+- Blocks and transactions produced by non-implementers MUST be validated by implementers using the same rules as with those coming from implementers, that is, the new rules.
+- Implementers MUST apply the proposed logic when evaluating transactions or computing their own data request eligibility.
+
+As a result:
+
+- Non-implementers MAY NOT get their blocks and transactions accepted by implementers.
+- Implementers MAY NOT get their valid blocks accepted by non-implementers.
+- Non-implementers MAY consolidate different blocks and superblocks than implementers.
+
+Due to this last point, this MUST be considered a consensus-critical protocol improvement. An adoption plan MUST be proposed, setting an activation date that gives miners enough time in advance to upgrade their existing clients.
+
+
+### Libraries, clients, and interfaces
+
+Most users of the RAD engine will be unaffected by this change, as it only changes the CBOR serialization of one error. For example, a client like [Sheikah][sheikah] will be unaffected because the data request result is encoded using JSON, so it can preserve the inner error in the `UnhandledIntercept` error case. Any clients that want to see the error message that would have been included in a `UnhandledIntercept` error can do so by fetching the reveals and running the tally script locally.
```

Comment:
```suggestion
Most users of the RAD engine will be unaffected by this change, as it only changes the CBOR serialization of one error. For example, a client like [Sheikah][sheikah] will not be affected because the data request result is encoded using JSON, so it can preserve the inner error in the `UnhandledIntercept` error case. Any clients that want to see the error message that would have been included in a `UnhandledIntercept` error can do so by fetching the reveals and running the tally script locally.
```